zeeguu/api/utils/route_wrappers.py

```diff
@@ -19,16 +24,37 @@ def with_session(view):
     def wrapped_view(*args, **kwargs):
         try:
             session_uuid = flask.request.args["session"]
-            session_object = Session.find(session_uuid)
-            if session_object is None:
+            user_id, timeout = SESSION_CACHE.get(
+                session_uuid,
+                (
+                    None,
+                    datetime.now() - timedelta(0, SESSION_CACHE_EXPIRE),
+                ),
+            )
+            if datetime.now() > timeout:
+                from zeeguu.api.endpoints.sessions import validate_session
+
+                print("----------- Updating Cache! -----------")
+                session_object = Session.find(session_uuid)
+                is_session_valid = validate_session(session_object)
```

Comment:
I was under the impression that we were only extending the session lifetime in the `/validate` endpoint. 

if we do it here, we'll end up just having to worry more on the frontend since every endpoint might invalidate the session. 

let's simplify the code and our lives.

Reply:
And that is correct. We only extend the session expiry date in the `/validate` edpoint. What you see here is just extending the timeout. 

This only invalidates the session in the case that a user has kept a session in for an entire month without ever refreshing it, and at that point I do think we should "re-validate" the session. This was the reason why I moved the logic of `is_session_valid` to a method, but realistically in the frontend I think we should aim for having a timeout that means that in normal circumstances, we never "re-validate" sessions, and we only do it after the server resets or something similar. This is also to ensue that we could force users to re-login by clearing the session table.